### ISSUE TITLE
Add a conflict resolution user interface

### ISSUE THREAD

**Issue by ckamm** [owncloud/client#5896]
Discussion started in #5894.

The client now notifies users about conflicts more persistently and has a view that can list all existing conflicts. It would be nice if the client could also help with the conflict resolution process.

To do that, we could open a conflict ui from a button in the conflict list. In most cases users will have to look at both versions of the file and manually merge differences.

The ui should have actions like:
* "Open folder" (similar to clicking the conflict list entry now)
* "Open synchronized server version"
* "Open unsynchronized local conflict copy"
* "The conflict is resolved, delete the conflict file"
* "The conflict is resolved, overwrite the file with the conflict file"
* "The conflict is resolved, keep both files renaming the conflict file"

**Comment by hodyroff** [comment]
Hm, lets discuss. As the counter request was to bring conflict files to the server - which for me would mean that resolution should also happen there?!

**Comment by ckamm** [comment]
@hodyroff Resolving conflicts on the server will always be limited by what you can do there. It's usually not just a "choose A or B" situation, but you need to open the files and manually merge to ensure you're not losing changes. The conflict could be an Excel table, Photoshop file, or anything else that two people changed at the same time and that might need specialized software to edit.

**Comment by michaelstingl** [comment]
FYI Here is the discussion about uploading conflict files: https://github.com/owncloud/client/issues/4557#issuecomment-307393122

Conflict caused by collaboration, should be made visible for all collaborators.

**Comment by ckamm** [comment]
Yes, uploading conflict files is in discussion - but I don't think it changes much about the conflict ui suggestion. We'll probably discuss both more for 2.5.

**Comment by ckamm** [comment]
@SamuAlfageme @mmattel I'm considering options here. Here is one of them:

![image](https://user-images.githubusercontent.com/833306/40431073-e9506c60-5ea6-11e8-897a-f564e5b6593c.png)

It's a pretty overwhelming dialog, unfortunately. Just context menu actions was even more confusing though, and I couldn't convey all the relevant info (like the date of the last change). 

Nothing is fixed here yet, feel free to suggest other designs or approaches!

**Comment by michaelstingl** [comment]
@ckamm Not sure if this approach makes things easier for regular users. Needs another thought. Higher priority for VFS, locking etc for now…

**Comment by mmattel** [comment]
As written in #6252, we need a UI for resolving conflicts. Conflicts can be existing and new.
I highly propose NOT to use the client as UI fo resolving conflicts - even this may look smart - it is not.

The majority of the users are ordinary office users and as often said, mostly not aware about a client doing all the job in the background. They are happy to know as less as possible and reluctant to think on a higher level.

**Comment by ckamm** [comment]
@mmattel What I was going for here is to have this dialog reachable from the file context menu as well as from the conflict list, so even users unaware of the client's existence would reach it.

Furthermore, conflicts are hard. Users must learn about them and how to resolve them or otherwise changes will get lost when conflicts are resolved badly or not at all. We're taking first steps about this by adding more documentation and linking it in relevant places. 

@michaelstingl I've removed this from the 2.6 milestone.
